core/shared/src/main/scala/com/lynbrookrobotics/potassium/streams/Stream.scala

```diff
@@ -214,37 +180,37 @@ abstract class Stream[+T] { self =>
     * @return a stream returning complete windows
     */
   def sliding(size: Int): Stream[Queue[T]] = {
-    var last = Queue.empty[T]
-
-    val ret = new Stream[Queue[T]] {
-      private val parent = self
-      override val expectedPeriodicity = self.expectedPeriodicity
-      override val originTimeStream = self.originTimeStream.map { o =>
-        o.sliding(size).map(_.last)
-      }
-    }
+    new Stream[Queue[T]] {
+      var unsubscribe: Cancel = null
+      var last = Queue.empty[T]
 
-    val ptr = WeakReference(ret)
-
-    var cancel: Cancel = null
-    cancel = this.foreach { v =>
-      ptr.get match {
-        case Some(s) =>
+      override def subscribeToParents(): Unit = {
+        unsubscribe = self.foreach { v =>
           if (last.size == size) {
             last = last.tail
           }
 
           last = last :+ v
 
           if (last.size == size) {
-            s.publishValue(last)
+            this.publishValue(last)
```

Comment:
I realize this is for another pr, but we shouldn't be waiting for certain number of values before updating children. This might cause problems if we want do a large sliding window. 

Instead we should have a queue of options of just a queue that grows until it reaches size

Reply:
Yeah, that could probably be introduced as a sliding variant.